utils/firestore.js

```diff
@@ -2180,40 +2217,147 @@ const getParticipantSelection = async (filter) => {
-            await kitStatusCounterVariation('assigned', 'addressPrinted');
             return true;
         } else {
             return false;
-        } }
+        } 
+    }
     catch(error){
         return new Error(error);
     }
 }
+const confirmShippmentKit = async (shipmentData) => {
+    try {
+        const snapshot = await db.collection("kitAssembly").where('687158491', '==', shipmentData['687158491']).get();
+        if(snapshot.size > 0) {
+            const docId = snapshot.docs[0].id;
+            await db.collection("kitAssembly").doc(docId).update(
+                {
+                    '221592017': '277438316',
+                    '661940160': shipmentData['661940160']
+                })
+            const secondSnapShot = await db.collection("participants").where('173836415.266600170.mouthwash.687158491', '==', shipmentData['687158491']).get();
+            const secondSnapShotDocId = secondSnapShot.docs[0].id;
+            const prevParticipantObject = secondSnapShot.docs[0].data()[173836415][266600170]['mouthwash'];
+            await db.collection("participants").doc(secondSnapShotDocId).update(
+                { 
+                    '173836415': {
+                        '266600170': {
+                            ...prevParticipantObject,
+                            'mouthwash': {
+                                '221592017': '277438316',
+                                '661940160': shipmentData['661940160']
+                            }
+                        }
+                    }
+                })
+            return true
+            }
+        else {
+            return false
+        }
+    }
+    catch (error) {
+        return new Error(error);
+    }
+}
+
+
+const processVerifyScannedCode = async (id) => {
+    try {
+        const snapShot = await db.collection('kitAssembly').where('531858099', '==', id).where('221592017', '==', '241974920').get();
+        if (snapShot.docs.length === 1) {
+            return { valid: true, UKID: snapShot.docs[0].data()[687158491] }
+        }
+        else { return false }
+    } catch (error) {
+        return new Error(error);
+    }
+}
+
+const storeKitReceipt = async (package) => {
+    try {
+        const snapshot = await db.collection("kitAssembly").where('972453354', '==', package['972453354']).get();
+        if(snapshot.size > 0) {
+            const docId = snapshot.docs[0].id;
+            const Connect_ID = snapshot.docs[0].data()['Connect_ID']
+            await db.collection("kitAssembly").doc(docId).update(
+                {
+                    '221592017': '375535639',
+                    '826941471': package['826941471'],
+                    '633640710': package['633640710']
+                })
+            const secondSnapShot = await db.collection("participants").where('173836415.266600170.mouthwash.687158491', '==', snapshot.docs[0].data()[687158491]).get();
+            const secondSnapShotDocId = secondSnapShot.docs[0].id;
```

Comment:
secondSnapShot and secondSnapShotDocId is not a very descriptive variable names and it would impove the readability to reference them like participantDocId or participantSnapShot